grpc-native/src/main/java/org/ballerinalang/net/grpc/builder/syntaxtree/SyntaxTreeGenerator.java

```diff
@@ -0,0 +1,351 @@
+                if (valueType.getValue()) {
+                    moduleMembers = moduleMembers.add(getValueTypeStream(valueType.getKey()).getTypeDefinitionNode());
+                }
+                moduleMembers = moduleMembers.add(getValueType(valueType.getKey()).getTypeDefinitionNode());
+            }
+        }
+
+        for (java.util.Map.Entry<String, Message> message : stubFile.getMessageMap().entrySet()) {
+            for (ModuleMemberDeclarationNode messageNode : getMessageNodes(message.getValue())) {
+                moduleMembers = moduleMembers.add(messageNode);
+            }
+        }
+
+        for (EnumMessage enumMessage : stubFile.getEnumList()) {
+            moduleMembers = moduleMembers.add(getEnum(enumMessage).getEnumDeclarationNode());
+        }
+
+        // ROOT_DESCRIPTOR
+        if (stubFile.getRootDescriptor() != null) {
+            Constant rootDescriptor = new Constant("string", "ROOT_DESCRIPTOR", stubFile.getRootDescriptor());
+            moduleMembers = moduleMembers.add(rootDescriptor.getConstantDeclarationNode());
+        }
+
+        // getDescriptorMap function
+        if (stubFile.getDescriptors().size() > 0) {
+            Function getDescriptorMap = new Function("getDescriptorMap");
+            getDescriptorMap.addReturns(
+                    getParameterizedTypeDescriptorNode("map", SYNTAX_TREE_VAR_STRING)
+            );
+            Map descriptorMap = new Map();
+            for (java.util.Map.Entry<String, String> descriptor : getSortedDescriptorMap(stubFile).entrySet()) {
+                descriptorMap.addStringField(descriptor.getKey(), descriptor.getValue());
+            }
+            getDescriptorMap.addReturnStatement(descriptorMap.getMappingConstructorExpressionNode());
+            getDescriptorMap.addQualifiers(new String[]{"isolated"});
+            moduleMembers = moduleMembers.add(getDescriptorMap.getFunctionDefinitionNode());
+        }
+
+        Token eofToken = AbstractNodeFactory.createIdentifierToken("");
+        ModulePartNode modulePartNode = NodeFactory.createModulePartNode(imports, moduleMembers, eofToken);
+        TextDocument textDocument = TextDocuments.from("");
+        SyntaxTree syntaxTree = SyntaxTree.from(textDocument);
+        return syntaxTree.modifyWith(modulePartNode);
+    }
+
+    public static SyntaxTree generateSyntaxTreeForServiceSample(ServiceStub serviceStub, boolean addListener) {
+        NodeList<ModuleMemberDeclarationNode> moduleMembers = AbstractNodeFactory.createEmptyNodeList();
+        ImportDeclarationNode importForGrpc = Imports.getImportDeclarationNode("ballerina", "grpc");
+        NodeList<ImportDeclarationNode> imports = AbstractNodeFactory.createNodeList(importForGrpc);
+
+        if (addListener) {
+            Listener listener = new Listener(
+                    "ep",
+                    getQualifiedNameReferenceNode("grpc", "Listener"),
+                    getImplicitNewExpressionNode(new String[]{"9090"}),
```

Comment:
What is the behavior when someone has more than one service in a single proto file?. Do all the services attach to a single listener with port 9090?.

Reply:
Yes. All the services are attached to a single listener with port 9090. Therefore we only add the listener to the sample_service_file which contains the first service of the proto file.